fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -233,69 +237,60 @@ protected RdfStream getResourceTriples(final int limit) {
 
         final LdpPreferTag ldpPreferences = new LdpPreferTag(returnPreference);
 
-        final RdfStream rdfStream = new RdfStream();
-
         final Predicate<Triple> tripleFilter = ldpPreferences.prefersServerManaged() ? x -> true :
             IS_MANAGED_TRIPLE.negate();
 
-        if (ldpPreferences.prefersServerManaged()) {
-            rdfStream.concat(getTriples(LdpRdfContext.class));
-        }
-
-        rdfStream.concat(filter(getTriples(TypeRdfContext.class), tripleFilter::test));
+        final List<Stream<Triple>> streams = new ArrayList<>();
 
-        rdfStream.concat(filter(getTriples(PropertiesRdfContext.class), tripleFilter::test));
+        streams.add(getTriples(PROPERTIES).filter(tripleFilter));
 
-        if (!returnPreference.getValue().equals("minimal")) {
+        if (returnPreference.getValue().equals("minimal")) {
+            if (ldpPreferences.prefersServerManaged()) {
+                streams.add(getTriples(SERVER_MANAGED)
+                    .filter(x -> x.getPredicate().equals(type) &&
+                            x.getObject().getURI().startsWith(LDP_NAMESPACE)));
+            }
+        } else {
 
             // Additional server-managed triples about this resource
             if (ldpPreferences.prefersServerManaged()) {
-                rdfStream.concat(getTriples(AclRdfContext.class));
-                rdfStream.concat(getTriples(RootRdfContext.class));
-                rdfStream.concat(getTriples(ContentRdfContext.class));
-                rdfStream.concat(getTriples(ParentRdfContext.class));
+                streams.add(getTriples(SERVER_MANAGED));
             }
 
             // containment triples about this resource
             if (ldpPreferences.prefersContainment()) {
-                rdfStream.concat(getTriples(ChildrenRdfContext.class).limit(limit));
+                if (limit == -1) {
+                    streams.add(getTriples(LDP_CONTAINMENT));
+                } else {
+                    streams.add(getTriples(LDP_CONTAINMENT).limit(limit));
+                }
             }
 
             // LDP container membership triples for this resource
             if (ldpPreferences.prefersMembership()) {
-                rdfStream.concat(getTriples(LdpContainerRdfContext.class));
-                rdfStream.concat(getTriples(LdpIsMemberOfRdfContext.class));
+                streams.add(getTriples(LDP_MEMBERSHIP));
             }
 
-            // Embed all hash and blank nodes
-            // using IS_MANAGED_TRIPLE directly to avoid Prefer header logic (we never want them for hash fragments)
-            rdfStream.concat(filter(getTriples(HashRdfContext.class), IS_MANAGED_TRIPLE.negate()::test));
```

Comment:
It appears that you are changing the logic to include `HashRdfContext` and `SkolemNodeRdfContext` in both `Prefer=minimal` and `!Prefer=minimal`, whereas they previously were only included in `!Prefer=minimal`.

Note: Hash and Skolem are being added here: https://github.com/fcrepo4/fcrepo4/pull/962/files#diff-d372035f89c8d0846920e20bd13e0145R245